lesson9/main.js

```diff
@@ -0,0 +1,55 @@
+//課題の配列
+const data = [
+   {to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク", src: "img/bookmark.jpeg"},
+   {to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ", src: "img/message.jpeg"}
+];
+
+const body = document.getElementById("body");
+const wrapper = document.createElement("div");
+body.appendChild(wrapper);
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => wrapper.remove();
+
+
+const createListElements = () => {
+   const fragment = document.createDocumentFragment();
+   const lists = document.getElementById("lists");
+   data.forEach((value) => {
+      const li = document.createElement("li");
+      const anchor = document.createElement("a");
+      const image = document.createElement("img");
+      const text = document.createTextNode(value.text);
+      anchor.href = `/${value.to}`;
+      image.alt = value.alt;
+      image.src = value.src;
+      anchor.appendChild(text);
+      anchor.appendChild(image);
+      li.appendChild(anchor);
+      fragment.appendChild(li);
+   });
+
+   lists.appendChild(fragment);
+}
+
+const waitTime = () => {
+   return new Promise (resolve => {
+      setTimeout(() => {
+         resolve(data);
+      }, 3000);
+   })
+}
+
+async function callCreateListElements() {
+   startLoading();
+   const value = await waitTime();
+   endLoading();
+   createListElements(value);
+}
```

Comment:
you should change function name. because `callCreateListElement` expect to create ListElement, but

Reply:
1, 2. I named it asyncProcessing, because these processes are asynchronous processing. Would it be better to give it a more specific name?
3. I referred to [this article](https://qiita.com/KeithYokoma/items/2193cf79ba76563e3db6).
96b2fdb

Reply:
Your code look like better than previous.

Reply:
thank you very much :)